New Docs Content: Button alignment, order, and grouping (#1092)
hashicorp/design-system: website/docs/patterns/button-alignment/partials/specifications/order.md

## Order

When performing an action or function, users generally think from highest to lowest priority and from positive to negative. How buttons and actions are ordered should respect this by ordering the primary (most important) action first, secondary action second, and if applicable, tertiary action last.

!!! Do

> Order buttons based on priority starting with the **highest** priority  action(usually by using a primary button) to **lowest** priority (either secondary or tertiary actions).

Reviewer: ```suggestion
Order buttons based on priority starting with the **highest** priority action (usually by using a primary button) to **lowest** priority (either secondary or tertiary actions).
```

!!!

### Conversational messaging

Users often expect to communicate with machines and applications as if they were humans. Adhering to these guidelines can help to reinforce this paradigm by encouraging users through a human-like conversational style.

![Button order through conversational style](/assets/patterns/button-alignment/order-conversational-example.png)

Consider the following scenario:

A user has taken an action to edit the settings of a cluster. Approached through a conversational method, the highest priority positive action (Save changes) comes first, with less-important and negative actions following.

An often depicted alternative to this scenario is to **end** with the most important action as seen in the following comparative example. When analyzed using a reading pattern, the example is awkward when read out loud and forces the user to read through items of lower importance first.

!!! Dont

#### Non-natural conversation based on priority

![Suboptimal grouping](/assets/patterns/button-alignment/suboptimal-grouping.png)

- User edits the settings of a cluster
    - "Would you like to learn more about this action?" "No"
    - "Would you like to cancel?" "No"

>     - "Would you like to save the changes to this cluster?" "Yes, finally that's what I was trying to do all along."

Reviewer: ```suggestion
    - "Would you like to save the changes to this cluster?" "Yes, finally that’s what I was trying to do all along."
```

!!!

!!! Do

> #### Nautral conversation based on priority

Reviewer 2: ```suggestion
#### Natural conversation based on priority
```

![Optimal order](/assets/patterns/button-alignment/optimal-ordering.png)

- User edits the settings of a cluster
    - "Would you like to save the changes to this cluster?" "Yes, thank you."
!!!

#### Exceptions

> An possible exception to this ordering method is in multi-step and progressive flows in which buttons fulfill both a functional purpose (create or edit something) as well as a navigational purpose (proceed, go back).

Reviewer: ```suggestion
A possible exception to this ordering method is in multi-step and progressive flows in which buttons fulfill both a functional purpose (create or edit something) as well as a navigational purpose (proceed, go back).
```
